ee/tabby-ui/components/chat.tsx

```diff
@@ -206,4 +209,40 @@ function ChatRenderer(
   )
 }
 
+function mergeMessagesByRole(body: BodyInit | null | undefined) {
```

Comment:
just move this to `use-patch-fetch.ts`? since tabby's backend always have this behavior (no consecutive same role message)